pxd.c

```diff
@@ -993,7 +994,8 @@ static void pxd_rq_fn(struct request_queue *q)
 			break;
 
 		/* Filter out block requests we don't understand. */
-		if (BLK_RQ_IS_PASSTHROUGH(rq)) {
+		if (BLK_RQ_IS_PASSTHROUGH(rq) ||
+				(!fc->allow_disconnected && !fc->connected)) {
```

Comment:
early return based on allow_disconnected and connected state. This ensures no allocation and immediate freeing of requests happen later.